src/transformers/modeling_utils.py

```diff
@@ -797,15 +811,19 @@ def _generate_beam_search(
         beam_scores = beam_scores.view(-1)  # shape (batch_size * num_beams,)
 
         # cache compute states
-        pasts = None  # self.prepare_pasts()
+        past = None
```

Comment:
XLM actually needs to have a cache dict initialized to use the cache.
But we have very few XLM models which are usable for generation (only the small `xlm-clm` model) so let's not bother too much with them for now.